src/dialog-user/services/dialogData.spec.ts

```diff
@@ -144,4 +144,17 @@ describe('DialogDataService test', () => {
     const expectedSortedResult = [[5, 'C'], [0, 'B'], [2, 'A']];
     expect(testSortedDescription).toEqual(expectedSortedResult);
   });
+  it('should allow a numeric Description field to be sorted in a dropdown', () => {
+    const testDropDownDescription = {
+      values: [
+        [0, '0'],
```

Comment:
Also, can you change the keys and values to something different?

It is impossible to tell if this really works or if it sorts by value instead :).

Reply:
Oh, and also without an `11` or something like that, it doesn't actually tell us if those are sorted as numbers or as strings..

Reply:
Also, we need to set the `data.data_type` to `integer` which is part of the original problem that javascript was sorting 1, 11, and 2 as if they were strings.

Reply:
Actually that wasn't the problem this PR addresses.  This deals with a use case where a user would decide to actually use numbers in the description field and then decide they want to sort the dropdown by the description.

Reply:
@eclarizio , I went ahead and updated one of the test cases to actually specify the data type.  It wasn't directly related to this test but I added it in for clarity sake.